src/common/code_environment.cc

```diff
@@ -115,7 +116,7 @@ int get_process_name(char *buf, int len)
 
 std::string get_process_name_cpp()
 {
-  char buf[32];
+  char buf[PATH_MAX];
```

Comment:
i think mingw-w64 supports `pthread_getname_np()`, see https://github.com/Alexpux/mingw-w64/blob/master/mingw-w64-libraries/winpthreads/src/thread.c#L1807. if #39570  gets merged, can we drop the `GetModuleFileNameA()` based implementation?

Reply:
Thanks for the suggestion, but it's not the same thing. Instead of retrieving the executable name, this will retrieve an assigned thread name. For example, the logging thread will just use the "log" name.

[1] https://github.com/ceph/ceph/blob/master/src/log/Log.cc#L407